sampler_remote_test.go

```diff
@@ -599,3 +601,50 @@ func TestRemotelyControlledSampler_printErrorForBrokenUpstream(t *testing.T) {
 	sampler.UpdateSampler()
 	assert.Contains(t, logger.String(), "failed to fetch sampling strategy:")
 }
+
+func TestRemotelyControlledSampler_samplingStrategiesFile(t *testing.T) {
+	content := []byte("probabilistic")
+	tmpfile, err := ioutil.TempFile("", "jaeger-client-go-strategies.json")
+	require.NoError(t, err)
+
+	filename := tmpfile.Name()
+	defer os.Remove(filename) // clean up
+
+	_, err = tmpfile.Write(content)
+	require.NoError(t, err)
+	err = tmpfile.Close()
+	require.NoError(t, err)
+
+	sampler := NewRemotelyControlledSampler(
+		"client app",
+		SamplerOptions.SamplingStrategiesFile(filename),
+	)
+	defer sampler.Close() // stop timer-based updates, we want to call them manually
+
+	assert.Equal(t, sampler.samplerOptions.samplingStrategiesFile, filename)
+
+	_, ok := sampler.Sampler().(*ProbabilisticSampler)
```

Comment:
I think this does not prove that the sampler was actually instantiated per the config file, because Remote defaults to probabilistic anyway. I think there is a way to pass a different init sampler to Remote one, or you can vary which strategy is provided in the config file.